### PR TITLE
 feat(pipeline): join processor

### PR DESCRIPTION
**Pull request by shuiyisong**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

Porting https://github.com/GreptimeTeam/pipeline/pull/44

Implement [join processor](https://www.elastic.co/guide/en/elasticsearch/reference/current/join-processor.html)

## Checklist

- [ ] I have written the necessary rustdoc comments.
- [x] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.

### CODE REVIEW

File: src/pipeline/src/etl/processor/join.rs [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+use crate::etl::field::{Field, Fields};
+use crate::etl::processor::{
+    yaml_bool, yaml_field, yaml_fields, yaml_string, Processor, FIELDS_NAME, FIELD_NAME,
+    IGNORE_MISSING_NAME, SEPARATOR_NAME,
+};
+use crate::etl::value::{Array, Map, Value};
+
+pub(crate) const PROCESSOR_JOIN: &str = "join";
+
+/// only support string array value
```
Comment (killme2008) [inline src/pipeline/src/etl/processor/join.rs:24]: Looks like it doesn't only support string array, but all the values which implement `to_str_value`.

File: src/pipeline/src/etl/processor/join.rs [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+#[derive(Debug, Default)]
```
Comment (killme2008) [inline src/pipeline/src/etl/processor/join.rs:25]:
```suggestion
/// A processor to join each element of an array into a single string using a separator string between each element
#[derive(Debug, Default)]
```

File: src/pipeline/tests/join.rs [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use greptime_proto::v1::value::ValueData::StringValue;
+use greptime_proto::v1::{ColumnDataType, ColumnSchema, SemanticType};
+use pipeline::{parse, Content, GreptimeTransformer, Pipeline, Value};
+
+#[test]
+fn test_simple_join() {
+    let input_value_str = r#"
```
Comment (killme2008) [inline src/pipeline/tests/join.rs:21]: Add a test to join other types, such as integers.

**Review by killme2008 — APPROVED** [review]
LGTM

**Review by paomian — APPROVED** [review]

**Pull request merged**